staging/src/k8s.io/kms/internal/mock_ratelimit_remote_service.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/kms/pkg/service"
+)
+
+type mockRateLimitRemoteService struct {
+	latencyRemoteService service.Service
+	limiter              flowcontrol.RateLimiter
+}
+
+var _ service.Service = &mockRateLimitRemoteService{}
+
+func (s *mockRateLimitRemoteService) Decrypt(ctx context.Context, uid string, req *service.DecryptRequest) ([]byte, error) {
+	if s.limiter.TryAccept() {
+		return s.latencyRemoteService.Decrypt(ctx, uid, req)
+	}
+	return nil, fmt.Errorf("remote decrypt rate limit exceeded")
+}
+
+func (s *mockRateLimitRemoteService) Encrypt(ctx context.Context, uid string, data []byte) (*service.EncryptResponse, error) {
+	if s.limiter.TryAccept() {
+		return s.latencyRemoteService.Encrypt(ctx, uid, data)
+	}
+	return nil, fmt.Errorf("remote encrypt rate limit exceeded")
+}
+
+func (s *mockRateLimitRemoteService) Status(ctx context.Context) (*service.StatusResponse, error) {
+	if s.limiter.TryAccept() {
+		return s.latencyRemoteService.Status(ctx)
+	}
+	return nil, fmt.Errorf("remote status rate limit exceeded")
```

Comment:
```suggestion
	return s.latencyRemoteService.Status(ctx)
```

same comment as above (no rate limiting or added latency for `Status` calls)